sites/chalice/app.py

```diff
@@ -0,0 +1,35 @@
+import copy
+import os
+import sys
+import json
+import chalice
+import bisect
+
+pkg_root = os.path.abspath(os.path.join(os.path.dirname(__file__)))  # noqa
+sys.path.insert(0, pkg_root)  # noqa
+
+from sites.controller import FusilladeConfig, GitlabController, FileController
+
+
+config = FusilladeConfig()
+gitlab = GitlabController()
+
+
+def get_groups():
+    group_file_path = "config/groups.json"
+    resp = gitlab.get_file_from_repo(group_file_path)
+    group_data = json.loads(resp.text).get('groups')
+    groups_in_file = [k for k,v in group_data if k != "user_default"]
+    return groups_in_file
+
+
+def add_user_to_group(service_account:str, groups:list):
+    group_file_path = "config/groups.json"
+    resp = gitlab.get_file_from_repo(group_file_path)
+    groups_file = FileController(resp.text, group_file_path)
+    # modify groups.json file thats committed currently
+    groups_file.updated_data = groups_file.add_user_to_groups(service_account, groups)
+    # create new branch, push changes
+    gitlab.create_branch(service_account)
+    gitlab.commit_changes_(service_account,groups_file)
```

Comment:
Need one more step here, which is to open a pull request (which Gitlab calls a "merge request"). Link to Gitlab API for opening a merge request: https://docs.gitlab.com/ee/api/merge_requests.html#create-mr

This will need the following info:
- project id
- source branch (get from `create_branch` call above)
- target branch (depends on stage)

Other helpful info to provide:
- description (stating this PR is automatically generated)
- approvals_before_merge (to require approval of the PR before it is merged in)
- remove_source_branch (so old branch is deleted when PR is merged in)